pex/resolver.py

```diff
@@ -95,6 +126,401 @@ def parsed_platform(platform=None):
   return Platform.create(platform) if platform and platform != 'current' else None
 
 
+class AtomicDirectory(namedtuple('AtomicDirectory', ['work_dir', 'target_dir'])):
+  @classmethod
+  def for_target_dir(cls, target_dir):
+    return cls(work_dir='{}.{}'.format(target_dir, uuid4().hex), target_dir=target_dir)
+
+  @property
+  def is_finalized(self):
+    return os.path.exists(self.target_dir)
+
+  def finalize(self):
+    if self.is_finalized:
+      return
+
+    try:
+      os.rename(self.work_dir, self.target_dir)
```

Comment:
The PR description mentioned us exploiting a property of Posix for this to work - would be good to memorialize that here.

Reply:
Done.